compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -530,10 +530,9 @@ static LogicalResult setWarpReductionConfig(func::FuncOp entryPoint,
 }
 
 /// Returns true if the index map represents a transpose that benefits from
-/// shared mem. Currently supports 2D transposes.
+/// shared mem.
 static bool isSharedMemTranspose(AffineMap indexMap) {
-  if (!indexMap.isEmpty() && indexMap.isPermutation() &&
-      indexMap.getNumInputs() == 2) {
+  if (!indexMap.isEmpty() && indexMap.isPermutation()) {
```

Comment:
I assume we want to limite to up to 3 parallel loops? I don't think the code generated would be efficient otherwise

Reply:
Not sure I follow? Right now we stick to linalg ops that don't have reduction loops (meaning all loops are parallel). 

If there are more than 3 parallel loops, the `setTransposeConfig` will assign a tile size of 1 to those loops such that we only care about the faster moving dimensions? 

Also, irrc you mentioned any higher order operation is flattened to 3 loops before this point anyway?

Reply:
> If there are more than 3 parallel loops, the setTransposeConfig will assign a tile size of 1 to those loops such that we only care about the faster moving dimensions?

I believe we still have some problem distributing more than 3 dimensions so that won't do what you want in this case. I could be wrong but we would at least need to add a test to confirm. If we have 4D and tile 1, 1, 32, 32 I'm not sure if the first dimension is the one that won't be distributed. If this is the case then it is fine.

> Also, irrc you mentioned any higher order operation is flattened to 3 loops before this point anyway?

That's the goal but this is not done right now.

Reply:
Ok. I've added a check to limit it to just 3 dims for now. I'll consider higher dims down the line.